pkg/trace/transform/transform.go

```diff
@@ -210,14 +218,21 @@ func MarshalEvents(events ptrace.SpanEventSlice) string {
 			str.WriteString(`"attributes":{`)
 			j := 0
 			e.Attributes().Range(func(k string, v pcommon.Value) bool {
-				if j > 0 {
-					str.WriteString(",")
+				// collect the attribute only if the key is json-parseable, else drop the attribute
+				if key, err := json.Marshal(k); err == nil {
+					if j > 0 {
+						str.WriteString(",")
+					}
+					str.WriteString(string(key))
+					str.WriteString(":")
+					if val, err := json.Marshal(v.AsString()); err == nil {
+						str.WriteString(string(val))
+					} else {
+						log.Warnf("Trouble parsing the following attribute value, dropping: %v", v.AsString())
+					}
+				} else {
+					log.Errorf("Error parsing the following attribute key on span event %v, dropping attribute: %v", e.Name(), k)
```

Comment:
Note that I couldn't come up with a way to test the resulting string when json.Marshal fails; lmk if you can think of anything.

Reply:
What about an empty string or a nil pointer? Do those end up hitting this code path?

Reply:
looks like passing something dumb might get it to fail:
https://stackoverflow.com/questions/33903552/what-input-will-cause-golangs-json-marshal-to-return-an-error

Reply:
Yes, but to test line 236 (or any other lines handling json marshaling errors), the event attribute would've needed to be created with this "unmarshable" value. And I don't think that's possible AFAICT (I tried to create an event with attribute value math.Inf(1) and the attribute was completely dropped from the event, like by otel, pre-marshaling).
So this code is really just a safeguard, that *probably* won't ever be run.. 🤔

Reply:
makes sense to me